raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -2763,78 +2765,396 @@ public void testDescribeQuorumNonLeader(boolean withKip853Rpc) throws Exception
     }
 
     @ParameterizedTest
-    @ValueSource(booleans = { true, false })
-    public void testDescribeQuorum(boolean withKip853Rpc) throws Exception {
-        int localId = randomReplicaId();
-        ReplicaKey closeFollower = replicaKey(localId + 2, withKip853Rpc);
-        ReplicaKey laggingFollower = replicaKey(localId + 1, withKip853Rpc);
-        Set<Integer> voters = Utils.mkSet(localId, closeFollower.id(), laggingFollower.id());
+    @CsvSource({ "true,0", "true,1", "false,0", "false,1" })
+    public void testDescribeQuorumOld(boolean withKip853Rpc, short apiVersion) throws Exception {
+        int localId = 0;
+        ReplicaKey local = replicaKey(localId, withKip853Rpc);
+        ReplicaKey follower1 = replicaKey(1, withKip853Rpc);
+        Set<Integer> voters = Utils.mkSet(localId, follower1.id());
+        VoterSet voterSet = VoterSetTest.voterSet(Stream.of(local, follower1));
+
+        RaftClientTestContext.Builder builder = new RaftClientTestContext.Builder(localId, local.directoryId().orElse(ReplicaKey.NO_DIRECTORY_ID))
+            .withStaticVoters(voters)
+            .withKip853Rpc(true);
```

Comment:
Why this is always set to `true`? Don't you wan to set it to `withKip853Rpc`? If you do that, I am sure you can remove the `apiVersion` parameter.

Reply:
`true` is a typo, thanks for catching that

I want this test to check the local leader is able to handle older versions of describe quorum requests (versions 0 and 1) no matter if it supports kip853Rpc or not

Reply:
> I want this test to check the local leader is able to handle older versions of describe quorum requests (versions 0 and 1) no matter if it supports kip853Rpc or not

Then call `withKip853Rpc(false)`. Please take a loot at the implementation but that flag basically determines which version of the RPCs to send.

Reply:
I'm aware that `withKip853Rpc` defines the versions of RPCs to send. The distinction I'm trying to make is `withKip853Rpc` does more than define what RPC versions to send in the test, it also defines what the local replica supports (for instance, if directory ids are supported) which can be different than what other replicas support.